### PR TITLE
PM-14547: Remove soft logged out account

### PR DESCRIPTION
**Pull request by matt-livefront**
## 🎟️ Tracking

[PM-14547](https://bitwarden.atlassian.net/browse/PM-14547)

## 📔 Objective

When long-pressing on a logged out account in the profile switcher, the remove option should be displayed instead of log out.

## 📸 Screenshots

https://github.com/user-attachments/assets/2de7e812-90b2-427d-ac82-4fe2eb7911f8

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-14547]: https://bitwarden.atlassian.net/browse/PM-14547?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by fedemkr — DISMISSED** [review]

File: BitwardenShared/UI/Auth/Extensions/AlertAuthTests.swift [hunk 1]
```diff
@@ -4,11 +4,13 @@ import XCTest
 
 class AlertAuthTests: BitwardenTestCase {
     /// `accountOptions(_:lockAction:logoutAction:)`
```
Comment (ezimet-livefront) [inline BitwardenShared/UI/Auth/Extensions/AlertAuthTests.swift:6]:
```suggestion
    /// `accountOptions(_:lockAction:logoutAction:removeAccountAction)`
```

File: BitwardenShared/UI/Auth/Extensions/AlertAuthTests.swift [hunk 2]
```diff
@@ -18,6 +20,42 @@ class AlertAuthTests: BitwardenTestCase {
...
+
+        try await subject.tapAction(title: Localizations.logOut)
+        XCTAssertEqual(actions, [Localizations.logOut])
+        actions.removeAll()
+
+        try await subject.tapAction(title: Localizations.cancel)
+        XCTAssertTrue(actions.isEmpty)
+    }
+
+    /// `accountOptions(_:lockAction:logoutAction:)` shows the account options for a logged out account.
```
Comment (ezimet-livefront) [inline BitwardenShared/UI/Auth/Extensions/AlertAuthTests.swift:36]:
```suggestion
    /// `accountOptions(_:lockAction:logoutAction:removeAccountAction:)` shows the account options for a logged out account.
```

**Review by ezimet-livefront — APPROVED** [review]
⛏  just a nitpick, but looks good.

**Review by ezimet-livefront — APPROVED** [review]

File: BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherHandlerTests.swift [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        XCTAssertEqual(
+            subject.handleAuthEvents,
+            [
+                .action(.logout(userId: loggedOutAccount.userId, userInitiated: true)),
+            ]
+        )
+    }
+}
+
+class MockProfileSwitcherHandlerProcessor:
```
[the remark this answers is not in the document]
Comment (matt-livefront) [reply BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherHandlerTests.swift:232]: Sure, I can move it.

**Review by KatherineInCode — APPROVED** [review]

**Pull request merged**